### ISSUE TITLE
Handling Dates in Typechecker

### ISSUE THREAD

**Issue by Legacy25** [UBOdin/mimir#75]
Dates need to be handled in the Typechecker comparisons. Sqlite doesn't have a date class so it reports the type for date columns as strings in ResultSetIterator. So we also need to support string and date comparisons (example ```SHIPDATE <= DATE('1998-1-1')```). I have worked around this issue by changing the following code -

```
case Comparison((Gt | Gte | Lt | Lte), lhs, rhs) =>
				if(typeOf(lhs) != TDate && typeOf(rhs) != TDate) {
					Typechecker.assertNumeric(Typechecker.escalate(typeOf(lhs), typeOf(rhs), "Comparison"))
				}
				TBool
```

Wondering if adding a case like ```(TDate, TString) | (TString, TDate) | (TDate, TDate) => TDate``` to ```escalate``` and then adding something like an ```assertDate``` would be better.

**Comment by okennedy** [comment]
This should be fixed in #73 

ResultSetIterator now takes an (optional) explicit schema, and uses the types it provides when applicable.

**Comment by Legacy25** [comment]
Moving this issue to new ticket.
